generated/lib.es5.d.ts

```diff
@@ -8,6 +8,68 @@ type UnionToIntersection<T> = (
   ? F
   : unknown;
 
+/*
+type Item<A extends readonly unknown[], T> = number extends A["length"]
+  ? T | undefined
+  : A["length"] extends 0
+  ? undefined
+  : T;
+*/
```

Comment:
Originally use for the `pop` and `shift` methods, but later I found it abundant because
```ts
const foo = [42] as const;
foo.pop();
foo.pop();
```
will give the wrong type. Blocked by https://github.com/microsoft/TypeScript/issues/40562.

I should have deleted these lines 😢